plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/snowflake/TrinoSnowflakeCatalog.java

```diff
@@ -171,6 +172,14 @@ public List<TableInfo> listTables(ConnectorSession session, Optional<String> nam
                 .toList();
     }
 
+    @Override
+    public List<SchemaTableName> listIcebergTables(ConnectorSession session, Optional<String> namespace)
+    {
+        return listTables(session, namespace).stream()
+                .map(TableInfo::tableName)
+                .collect(toImmutableList());
+    }
+
```

Comment:
This commit should include a unit for the new API in `io.trino.plugin.iceberg.catalog.BaseTrinoCatalogTest`

Reply:
I added tests to testListTables previously extending the tests to cover cases with non-iceberg table relations.